### ISSUE TITLE
[$250] Track Expense - Hmm..not here page on selecting category with no existing workspace

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#43073]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4-79-3
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:** https://expensify.testrail.io/index.php?/tests/view/4594444&group_by=cases:section_id&group_order=asc&group_id=309130
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com
2. Create new account
3. Submit a track expense in the self DM
4. Click on Categorize it on the actionable whisper options of a track expense
5. Select a category shown in the list to categorize the expense

## Expected Result:
As per step TC 6 - After selecting a category, the "To" field should be filled with a "new" workspace

## Actual Result:
Hmm..not here page on selecting category with no existing workspace

## Workaround:
n/a

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![image](https://github.com/Expensify/App/assets/93399543/6da6eaf3-07ff-49af-883e-65e82a059f60)

![image](https://github.com/Expensify/App/assets/93399543/7ac0c49b-c8be-45f6-8d39-f4ca0663abaa)

https://github.com/Expensify/App/assets/93399543/1ddb7c4e-e7e8-4ddd-a07d-548b381b12de

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a18541e1775fab5a</li>
        <li>Upwork Job ID: 1798437262634635540</li>
        <li>Last Price Increase: 2024-06-05</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @mananjadhav</details>

**Comment by kbecciv** [comment]
We think that this bug might be related to #vip-vsb

**Comment by blimpich** [comment]
Looking into this.

**Comment by blimpich** [comment]
Reproduced it locally. I think [this PR](https://github.com/Expensify/App/pull/41344) is really suspicious given the timing and the area of the codebase. Reverting that PR locally doesn't fix it though, so that doesn't seem like its the root cause.

Not 100% sure if this is backend or frontend, seems like frontend because I can't find any wonky network requests or logs that are giving me any leads. Though maybe it has to do with bad handling of onyx updates in the backend. Not sure.

Reached out in #expensify-opensource ([link](https://expensify.slack.com/archives/C01GTK53T8Q/p1717541695825979))

**Comment by blimpich** [comment]
I think this might be a backend change, since I went back about a week and a half in App history and its still an issue. Stuck on how to debug this. Can't seem to figure out where it is going wrong.

**Comment by blimpich** [comment]
Huh, well when I actually try to reproduce this on staging it doesn't reproduce, so maybe someone put out a fix that is related to this.

**Comment by blimpich** [comment]
Weird, yup, pulled everything and now its working locally on main.

**Comment by blimpich** [comment]
[Asked QA to retest](https://expensify.slack.com/archives/C9YU7BX5M/p1717545177697899)

In the meantime I think we can demote as this is no longer reproducing for me on staging or locally. Should probably figure out what the exact cause was, but seems to be fixed.

**Comment by kavimuru** [comment]
Tester can reproduce the bug with new gmail account too.

https://github.com/Expensify/App/assets/43996225/fce8536a-f0c6-496f-b1f8-4af0e66e0a99

https://github.com/Expensify/App/assets/43996225/e50b1841-1b03-4d3d-be25-488ee1ee9eff

**Comment by blimpich** [comment]
@cretadn22 thank you for your proposal. Question: wouldn't the root cause of this problem be that the reportID is undefined when it should be defined?

**Comment by blimpich** [comment]
This to me seems like the issue, the backend isn't finding the report when we open the list of categories:

![Screenshot 2024-06-05 at 12 00 24 PM](https://github.com/Expensify/App/assets/17171674/e9f3fcde-fa60-445d-918b-a203814a29ba)

thank you @ZhenjaHorbach for pointing out this error to me [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1717613072346799?thread_ts=1717541695.825979&cid=C01GTK53T8Q).

**Comment by blimpich** [comment]
Yeah we can see here that we really shouldn't be calling openReport since (A) why would we need to when we're just opening the category list and (B) we don't call openReport when we do the same flow but with a workspace.

https://github.com/Expensify/App/assets/17171674/97f6793c-c908-4376-b3c2-838c060c6ec9

**Comment by blimpich** [comment]
Opening this up to contributors since [this discussion](https://expensify.slack.com/archives/C01GTK53T8Q/p1717541695825979) makes me believe this is a frontend issue.

**Comment by CortneyOfstad** [comment]
@getusha any feedback on the proposal above?

**Comment by mananjadhav** [comment]
Reviewing proposals.

**Comment by mananjadhav** [comment]
I am testing the proposals to see if it works and doesn't break anything. I'll have an update before EOD today.

**Comment by mananjadhav** [comment]
I was out sick past 3-4 days. I'll post an update today/tomorrow.

**Comment by CortneyOfstad** [comment]
Thanks @mananjadhav and hope you're feeling better!

**Comment by mananjadhav** [comment]
Apologies for the delay here. I've been trying to reproduce this on staging and the latest main. But I haven't been able to reproduce the bug. Earlier when it was reported I was able to, but not recently. @kbecciv @CortneyOfstad Are you able to consistently reproduce this?

**Here's the video from Web**

https://github.com/Expensify/App/assets/3069065/84973814-e53f-4201-8324-a0c87921e949

**iPhone Safari**

https://github.com/Expensify/App/assets/3069065/d1015c7b-5202-4a71-8182-9a6572e44f13


and I've tried this on Native, Mobile Web, Mac Safari. Tried it with 3 different new accounts. 

@CortneyOfstad I do see `Not found` when we click on the Workspace name. I am assuming that is expected as the workspace is not created yet. May be that shouldn't be clickable? But definitely a separate issue.
